content/en/docs/setup/production-environment/tools/kubeadm/setup-ha-etcd-with-kubeadm.md

```diff
@@ -80,21 +81,33 @@ this example.
     member running on it using the following script.
 
     ```sh
-    # Update HOST0, HOST1, and HOST2 with the IPs or resolvable names of your hosts
+    # Update HOST0, HOST1 and HOST2 with the IPs of your hosts
     export HOST0=10.0.0.6
     export HOST1=10.0.0.7
     export HOST2=10.0.0.8
 
+    # Update NAME0, NAME1 and NAME2 with the hostnames of your hosts
+    export NAME0="infra0"
+    export NAME1="infra1"
+    export NAME2="infra2"
+
     # Create temp directories to store files that will end up on other hosts.
     mkdir -p /tmp/${HOST0}/ /tmp/${HOST1}/ /tmp/${HOST2}/
 
-    ETCDHOSTS=(${HOST0} ${HOST1} ${HOST2})
-    NAMES=("infra0" "infra1" "infra2")
+    HOSTS=(${HOST0} ${HOST1} ${HOST2})
+    NAMES=(${NAME0} ${NAME1} ${NAME2})
 
-    for i in "${!ETCDHOSTS[@]}"; do
-    HOST=${ETCDHOSTS[$i]}
+    for i in "${!HOSTS[@]}"; do
+    HOST=${HOSTS[$i]}
     NAME=${NAMES[$i]}
     cat << EOF > /tmp/${HOST}/kubeadmcfg.yaml
+    apiVersion: "kubeadm.k8s.io/v1beta2"
+    kind: InitConfiguration
+    nodeRegistration:
+        name: ${NAME}
+    localAPIEndpoint:
+        advertiseAddress: ${HOST}
```

Comment:
the actual issue here is that if the generated certificates will not include the right names / IPs, because they are run on the same machine. including the InitConfiguration fields here fixes it.
https://github.com/kubernetes/kubeadm/issues/2334#issuecomment-868025463